### PR TITLE
Generate ci cd for eventstore

### PR DESCRIPTION
**Pull request by carlosrecuero**
**What this PR does / why we need it**:
There is ci/cd defined for EventStore

**Which issue(s) this PR fixes**:
#938

Please reference the issue this PR will close: #938

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by sungam3r2022** [comment]
You may also include new files here:
изображение

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        env:
+          EVENTSTORE_INSECURE: true
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        include-prerelease: false
+    - name: Restore dependencies
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:28]:
```suggestion
    - name: Restore
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      run: dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
+    - name: Build
+      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
+    - name: Test
+      run: dotnet test ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj --verbosity normal
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:33]:
```suggestion
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - name: dotnet pack
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:34]:
```suggestion
    - name: Pack
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      run: dotnet pack ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj -c $BUILD_CONFIG --include-source --include-symbols -o ./artifacts
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:35]:
```suggestion
      run: dotnet pack ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj -c $BUILD_CONFIG -o ./artifacts
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    - name: setup nuget
+      uses: NuGet/setup-nuget@v1.0.5
+      env:
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+      with:
+        nuget-version: latest
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:41]:
```suggestion
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Publish EventStore Health Check nuget
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:42]:
```suggestion
    - name: Publish
```

File: .github/workflows/healthchecks_eventstore_cd_preview.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    - name: dotnet pack
+      run: dotnet pack ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj --version-suffix $VERSION_SUFFIX -c $BUILD_CONFIG --include-source --include-symbols -o ./artifacts
+    - name: setup nuget
+      uses: NuGet/setup-nuget@v1.0.5
+      env:
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+      with:
+        nuget-version: latest
+    - name: Publish EventStore Health Check nuget
+      run: dotnet nuget push ./artifacts/AspNetCore.HealthChecks.EventStore.*.nupkg -k ${{secrets.NUGET_API_KEY}} -s https://api.nuget.org/v3/index.json --skip-duplicate
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd_preview.yml:44]:
```suggestion
    - name: Restore
      run: dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
    - name: Build
      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
    - name: Pack
      run: dotnet pack --no-build ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj --version-suffix $VERSION_SUFFIX -c $BUILD_CONFIG -o ./artifacts
    - name: Publish
      run: dotnet nuget push ./artifacts/AspNetCore.HealthChecks.EventStore.*.nupkg -k ${{secrets.NUGET_API_KEY}} -s https://api.nuget.org/v3/index.json --skip-duplicate
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    branches: [ master ]
+    paths:
+      - src/HealthChecks.EventStore/**
+      - test/HealthChecks.EventStore.Tests/**
+      - .github/workflows/healthchecks_eventstore_ci.yml
+      - Directory.Build.props
+      - Directory.Build.targets
+    tags-ignore:
+      - release-*
+  pull_request:
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:14]:
```suggestion

  pull_request:
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      BUILD_CONFIG: Release
+    runs-on: ubuntu-latest
+    services:
+      eventstore:
+        image: eventstore/eventstore
+        ports:
+          - "2113:2113"
+          - "1113:1113"
+        env:
+          EVENTSTORE_INSECURE: true
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:20]:
```suggestion
```

File: .github/workflows/healthchecks_eventstore_cd_preview.yml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      VERSION_SUFFIX: -rc2.${{ github.run_number }}
+    runs-on: ubuntu-latest
+    services:
+      eventstore:
+        image: eventstore/eventstore
+        ports:
+          - "2113:2113"
+          - "1113:1113"
+        env:
+          EVENTSTORE_INSECURE: true
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd_preview.yml:21]:
```suggestion
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+          EVENTSTORE_INSECURE: true
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        include-prerelease: false
+    - name: Restore dependencies
+      run: dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:42]:
```suggestion
    - name: Restore
      run: |
        dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj &&
        dotnet restore ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Check formatting
+      working-directory: ./src/HealthChecks.EventStore
+      run: |
+        dotnet tool install -g dotnet-format --version 6.0.243104 --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
+        dotnet format --no-restore --verify-no-changes --severity warn whitespace || (echo "Run 'dotnet format' to fix whitespace issues" && exit 1)
+        dotnet format --no-restore --verify-no-changes --severity warn analyzers || (echo "Run 'dotnet format' to fix analyzers issues" && exit 1)
+        dotnet format --no-restore --verify-no-changes --severity warn style || (echo "Run 'dotnet format' to fix style issues" && exit 1)
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:49]:
```suggestion
        dotnet format --no-restore --verify-no-changes --severity warn ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj || (echo "Run 'dotnet format' to fix issues" && exit 1) &&
        dotnet format --no-restore --verify-no-changes --severity warn ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj || (echo "Run 'dotnet format' to fix issues" && exit 1)
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Build
+      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:51]:
```suggestion
      run: |
        dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj &&
        dotnet build --no-restore ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Test
+      run: dotnet test ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj --verbosity normal
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:53]:
```suggestion
      run: dotnet test --no-restore --no-build ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+  build:
+    env:
+      BUILD_CONFIG: Release
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:18]:
```suggestion
        dotnet-version: 6.0.202
```

File: .github/workflows/healthchecks_eventstore_cd_preview.yml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    env:
+      BUILD_CONFIG: Release
+      VERSION_SUFFIX: -rc2.${{ github.run_number }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd_preview.yml:19]:
```suggestion
        dotnet-version: 6.0.202
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+          - "2113:2113"
+          - "1113:1113"
+        env:
+          EVENTSTORE_INSECURE: true
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:40]:
```suggestion
        dotnet-version: 6.0.202
```

**Comment by sungam3r** [comment]
Updated to align with other CI/CD worflows.

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.202
+        include-prerelease: false
+    - name: Restore
+      run: |
+        dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj &&
+        dotnet restore ./test/HealthChecks.EventStore.Tests/HealthChecks.EventStore.Tests.csproj
+    - name: Check formatting
+      working-directory: ./src/HealthChecks.EventStore
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:47]:
```suggestion
```

**Comment by sungam3r** [comment]
2 failed tests now - just as before my changes.

**Comment by NielsPilgaard** [comment]
PR #1500 should fix up this PR.
Additionally, this PR will resolve #875

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+
+jobs:
+  build:
+    env:
+      BUILD_CONFIG: Release
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:16]:
```suggestion
      uses: actions/setup-dotnet@v3
```

File: .github/workflows/healthchecks_eventstore_cd_preview.yml [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  build:
+    env:
+      BUILD_CONFIG: Release
+      VERSION_SUFFIX: -rc2.${{ github.run_number }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd_preview.yml:17]:
```suggestion
      uses: actions/setup-dotnet@v3
```

File: .github/workflows/healthchecks_eventstore_cd.yml [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 6.0.x
+    - name: Restore
+      run: dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
+    - name: Build
+      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd.yml:22]:
```suggestion
      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj -c $BUILD_CONFIG
```

File: .github/workflows/healthchecks_eventstore_cd_preview.yml [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 6.0.x
+    - name: Restore
+      run: dotnet restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
+    - name: Build
+      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_cd_preview.yml:23]:
```suggestion
      run: dotnet build --no-restore ./src/HealthChecks.EventStore/HealthChecks.EventStore.csproj -c $BUILD_CONFIG
```

File: .github/workflows/healthchecks_eventstore_ci.yml [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+        image: eventstore/eventstore
+        ports:
+          - "2113:2113"
+          - "1113:1113"
+        env:
+          EVENTSTORE_INSECURE: true
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
```
Comment (sungam3r) [inline .github/workflows/healthchecks_eventstore_ci.yml:38]:
```suggestion
      uses: actions/setup-dotnet@v3
```

File: test/HealthChecks.EventStore.Tests/Functional/EventStoreHealthCheckTests.cs [hunk 24]
```diff
@@ -29,14 +29,15 @@ public async Task be_healthy_if_eventstore_is_available_with_uri_format()
             response.StatusCode
                 .Should().Be(HttpStatusCode.OK);
         }
+
         [Fact]
         public async Task be_healthy_if_eventstore_is_available()
         {
             var webHostBuilder = new WebHostBuilder()
                 .ConfigureServices(services =>
                 {
```
Comment (sungam3r) [inline test/HealthChecks.EventStore.Tests/Functional/EventStoreHealthCheckTests.cs:38]:
```suggestion
                {
                    services.Configure<HealthCheckOptions>(opt => opt.ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse);
```

**Comment by NielsPilgaard** [comment]
The eventstore service container might need more env vars to work correctly here, I can investigate in a few hours. 

All tests passed in my PR, using EventStore in Docker, so I'm fairly certain I can find the right env vars.

**Pull request merged**